lib/puppet-syntax/hiera.rb

```diff
@@ -74,7 +74,8 @@ def check(filelist)
 
       filelist.each do |hiera_file|
         begin
-          yamldata = YAML.load_file(hiera_file)
+          yamlargs = RUBY_VERSION >= '3.1' ? { aliases: true } : {}
```

Comment:
Should this look at `Psych::VERSION` instead?

Reply:
It felt more hacky to pin to an internal library version vs the more top-level Ruby version

Reply:
Can't you update Psych via dependencies? Either way this is still pretty hacky and it's not going to get cleaner so I'm fine either way. I guess it's fine either way.

Reply:
I agree with @ekohl here. Ruby bundles a Psych version, but it can be updated. So we should check for the Psych version, not the Ruby version.

Reply:
Updated to be outside the loop and use Psych::VERSION